### ISSUE TITLE
StackOverflowError ClassWalker

### ISSUE THREAD

**Issue by paulcwarren** [paulcwarren/spring-content#845]
**Describe the bug**
StackOverflowError occurs when walking entity class to map content properties.

**Additional context**
Partial stack trace is:

`Caused by: java.lang.StackOverflowError at java.base/java.lang.reflect.Field.getAnnotation(Field.java:1142) at java.base/java.lang.reflect.AnnotatedElement.isAnnotationPresent(AnnotatedElement.java:274) at java.base/java.lang.reflect.AccessibleObject.isAnnotationPresent(AccessibleObject.java:517) at org.springframework.content.commons.mappingcontext.ContentPropertyBuilderVisitor.visitField(ContentPropertyBuilderVisitor.java:92) at org.springframework.content.commons.mappingcontext.ClassWalker.accept(ClassWalker.java:24)`

**Comment by paulcwarren** [comment]
Created this [repo](https://github.com/paulcwarren/issue-845) to try and reproduce.

**Comment by springframeworkguru** [comment]
@paulcwarren - created this gist of the full stack trace

https://gist.github.com/springframeworkguru/9c98b5cbfade28f391853a02e65184bb

**Comment by springframeworkguru** [comment]
@paulcwarren - See the PR below. I was able to recreate the issue. I updated to the Spring Boot version I'm using, Java 11, and the Spring Content deps, also added my full data model. 

fyi - It worked okay with my full model under your dependencies and Spring version. I think it wasn't until I used the same deps that I saw the error. Thanks!

https://github.com/paulcwarren/issue-845/pull/1
